x-pack/legacy/plugins/uptime/public/pages/settings.tsx

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import {
+  EuiForm,
+  EuiTitle,
+  EuiSpacer,
+  EuiDescribedFormGroup,
+  EuiFieldText,
+  EuiFormRow,
+  EuiCode,
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiToast,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { connect } from 'react-redux';
+import { isEqual } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { ChromeBreadcrumb } from 'kibana/public';
+import { AppState } from '../state';
+import { selectDynamicSettings } from '../state/selectors';
+import { DynamicSettingsState } from '../state/reducers/dynamic_settings';
+import { getDynamicSettings, setDynamicSettings } from '../state/actions/dynamic_settings';
+import { DynamicSettings, defaultDynamicSettings } from '../../common/runtime_types';
+import { PageHeader } from './page_header';
+
+interface Props {
+  dynamicSettingsState: DynamicSettingsState;
+}
+
+interface DispatchProps {
+  dispatchGetDynamicSettings: typeof getDynamicSettings;
+  dispatchSetDynamicSettings: typeof setDynamicSettings;
+}
+
+export const SettingsPageComponent = ({
+  dynamicSettingsState: dss,
+  dispatchGetDynamicSettings,
+  dispatchSetDynamicSettings,
+}: Props & DispatchProps) => {
+  useEffect(() => {
+    dispatchGetDynamicSettings('');
+  }, [dispatchGetDynamicSettings]);
```

Comment:
is it necessary to pass empty string?

Reply:
ah, good catch

Reply:
Yes, because createAction doesn't like actions without arguments for some reason.

Reply:
because you have defined action like this
`export const getDynamicSettings = createAction<{}>('GET_DYNAMIC_SETTINGS');`

just remove {} from generic type and it should be fine.